application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -125,6 +125,19 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
         String authorId = args.get(0);
         String messageId = args.get(1);
         String channelId = args.get(2);
+        ForumChannel helperForum = getHelperForum(event.getJDA());
+        TextChannel sourceChannel = event.getChannel().asTextChannel();
+
+        // to avoid race condition, when multiple users do the transfer.
+        // message is retrieved, already transferred messages are deleted.
+        event.getChannel()
+            .retrieveMessageById(messageId)
+            .queue(notHandled -> transferFlow(event, channelId, authorId, messageId),
+                    handled -> alreadyHandled(sourceChannel, helperForum));
```

Comment:
u can avoid the comments by writing more self-explanatory code. in this case, u could create a helper method named  `handleIfNotAlreadyHandled` or `handleIfUnhandled`.

whats tripping me, as reader, is the rather unconvential way this detects deletion (im not saying its wrong, but a normal user would expect some simple `if`).
so I think a comment would be helpful that says:
```java
// Has been handled if original message was deleted by now.
// Deleted messages cause retrieveMessageById to fail.
```

The only other comment I have is that right now u react on any failure of `retreiveMessageById` with "oh, so it was handled already". But that might not be the case, the method can also fail bc of other reasons.
The interface should give u the exception (likely that `handled` thing?) and JDA offers the failure-reason as enum somewhere. if u search the code base with `instanceof`, u should find examples.

Also, I find the use of `notHandled` and `handled` confusing. Ur naming these variables to indicate what the lambda is about, but by doing that, one completely doesnt understand what the variable itself is even for. Instead, name the lambda:
```java
Foo notHandledAction = any -> transferFlow(...);
Foo handledAction = any -> alreadyHandled(...);

event.getChannel()
  ...
  queue(notHandledAction, handledAction);
```

Reply:
Hmm i wasn't too happy with the handling either, you're right it could fail due to a lot of reasons, i'll start refactoring thanks.